### PR TITLE
Improved docstrings for Elemwise functions

### LINKED ISSUE #292: DOC: Docstrings for `Elemwise` `Op`s are not very helpful
### Issue with current documentation:

When a user checks a docstring for most basic `Elemwise` `Op`s, he is met with a wall of generic boilerplate:

```
import pytensor.tensor as pt

help(pt.exp)
```

Returns:

```
Help on Elemwise in module pytensor.tensor.math:

<pytensor.tensor.elemwise.Elemwise object>
    e^`a`
    
    Generalizes a scalar `Op` to tensors.
    
    All the inputs must have the same number of dimensions. When the
    `Op` is performed, for each dimension, each input's size for that
    dimension must be the same. As a special case, it can also be one
    but only if the input's `broadcastable` flag is ``True`` for that
    dimension. In that case, the tensor is (virtually) replicated
    along that dimension to match the size of the others.
    
    The dtypes of the outputs mirror those of the scalar `Op` that is
    being generalized to tensors. In particular, if the calculations
    for an output are done in-place on an input, the output type must
    be the same as the corresponding input type (see the doc of
    `ScalarOp` to get help about controlling the output type)
    
    Notes
    -----
    -``Elemwise(add)``: represents ``+`` on tensors ``x + y``
    -``Elemwise(add, {0 : 0})``: represents the ``+=`` operation ``x += y``
    -``Elemwise(add, {0 : 1})``: represents ``+=`` on the second argument ``y += x``
    -``Elemwise(mul)(np.random.random((10, 5)), np.random.random((1, 5)))``:
    the second input is completed along the first dimension to match the first input
    -``Elemwise(true_div)(np.random.random(10, 5), np.random.random(10, 1))``: same but along the
    second dimension
    -``Elemwise(int_div)(np.random.random((1, 5)), np.random.random((10, 1)))``:
    the output has size ``(10, 5)``.
    -``Elemwise(log)(np.random.random((3, 4, 5)))``
```

I'm not sure how much needs to be said on the exponential function as such, but there are some important `Elemwise` `Op`s that are not so obvious. For example, the fact that `pt.eq` needs to be used instead of `==` isn't immediately obvious, and could be explained with examples (and counter-examples) in its docstring. A small examples section would also a opportunity to present new users with some basics about building computational graphs, compiling functions, using `.eval()`, etc.

At the very least, the boilerplate is clutter that could be eliminated. The clutter is worse when you open the docstring in juypter, because it displays the Class docstring (shown above) and the `Call` docstring, which explains what an `Apply` node is. 

I remember being very confused by all this when I was getting started, so it might be something nice to refine.

### Idea or request for content:

_No response_

### PR DESCRIPTION
**Pull request by fonnesbeck**
## Description

Added more robust docstrings and removed the Elemwise docstring that was appended to each function.

## Related Issue

- [x] Closes #292 
- [ ] Related to #

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [x] Documentation
- [x] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

File: pytensor/tensor/math.py [hunk 1]
```diff
@@ -681,39 +681,32 @@ def largest(*args):
...
-@scalar_elemwise
 def ge(a, b):
-    """a >= b"""
+    """Elementwise greater-than-or-equal-to comparison, `a >= b`."""
 
 
-@scalar_elemwise
 def eq(a, b):
-    """a == b"""
+    """Elementwise equality comparison, `a == b`."""
```
Comment (ricardoV94) [inline pytensor/tensor/math.py:701]: Adding a special note for `eq` and `neq` which can't be overloaded (while >, >= and alike are overloaded just fine)

```suggestion
    """Elementwise equality comparison, `a == b`.
    
    Due to Python rules it is not possible to correcly overload the equality symbol `==`,
    so `eq` must always be used to compute the Elemwise equality of TensorVariables.
    """
```

File: pytensor/tensor/math.py [hunk 2]
```diff
@@ -681,39 +681,32 @@ def largest(*args):
...
 
 
-@scalar_elemwise
 def neq(a, b):
-    """a != b"""
+    """Elementwise inequality comparison, `a != b`."""
```
Comment (ricardoV94) [inline pytensor/tensor/math.py:705]:
```suggestion
    """Elementwise inequality comparison, `a != b`.
    
    Note
    ------
    Due to Python rules, it is not possible to correctly overload the non-equality symbol `!=`,
    so `neq` must always be used to compute the Elemwise non-equality of TensorVariables.
    """
```
    """
```

File: pytensor/tensor/math.py [hunk 3]
```diff
@@ -722,7 +715,25 @@ def isnan(a):
 
 
 def isnan(a):
-    """isnan(a)"""
+    """
+    isnan(a)
+
+    Elementwise test for NaN (not a number).
+
+    This function is a copy of the `isnan` function defined in the
```
Comment (ricardoV94) [inline pytensor/tensor/math.py:723]: I don't think we need this detail?
```suggestion
    This function is a copy of the `isnan` function defined in the
```

File: pytensor/tensor/math.py [hunk 4]
```diff
@@ -733,7 +744,21 @@ def isnan(a):
...
 def isinf(a):
-    """isinf(a)"""
+    """
+    isinf(a)
+
+    Elementwise test for infinite value.
+
+    Parameters
+    ----------
+    a : TensorVariable
```
Comment (ricardoV94) [inline pytensor/tensor/math.py:754]: All these should have TensorLike (I am not sure the capitalization) as inputs (anything that can be correctly converted to TensorVariables)
```suggestion
    a : TensorLike
```

File: pytensor/tensor/special.py [hunk 5]
```diff
@@ -485,6 +485,21 @@ def c_code_cache_version():
...
 def softmax(c, axis=None):
+    """
+    Compute the softmax of a vector along a specified axis.
+
+    Parameters
+    ----------
+    c : TensorVariable
+        The input tensor.
+    axis : int
+        The axis along which to compute the softmax.
```
Comment (ricardoV94) [inline pytensor/tensor/special.py:496]:
```suggestion
    axis : int or None
        The axis along which to compute the softmax. If None, softmax is applied along all axes.
```

**Comment by fonnesbeck** [comment]
Not clear why the fast-compile tests are failing here.

**Comment by ricardoV94** [comment]
> Not clear why the fast-compile tests are failing here.

Something must have actually broken. Perhaps the Elemwise decorator was doing something dumb that fails with the new docstrings? Have to investigate

**Comment by twiecki** [comment]
It seems by force-pushing you removed all commits @fonnesbeck. are they still to be found somewhere?

**Pull request closed without merging**